### PR TITLE
added age api and added class group field in class section

### PR DESCRIPTION
**Pull request by krishnakanth12**
please lookout and let me know any changes

### CODE REVIEW

File: eCampusAPI/ecampus_api/master/models.py [hunk 1]
```diff
@@ -63,6 +65,9 @@ class Caste(models.Model):
     created_on = models.DateTimeField(auto_now_add=True, null=True)
     created_by = models.IntegerField('created_by')
 
- 
-
+class Age(models.Model):
+    class_name = models.ForeignKey(ClassName, on_delete=models.PROTECT, related_name='choose_age_classname')
+    age = models.IntegerField('age',validators=[age_validator])
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/models.py:70]: remove age field and add from_age and to_age field.

File: eCampusAPI/ecampus_api/master/urls.py [hunk 2]
```diff
@@ -33,4 +33,9 @@
     path('caste/detail/<pk>/',CasteViewset.as_view({'get': 'retrieve'}), name='detail_caste'),
     path('caste/update/<pk>/',CasteViewset.as_view({'put': 'update'}), name='update_caste'),
     path('caste/delete/<pk>/',CasteViewset.as_view({'delete': 'destroy'}), name='delete_caste'),
+    path('age/create/',AgeViewset.as_view({'post': 'create'}), name='create_age'),
+    path('age/list/',AgeViewset.as_view({'get': 'list'}), name='list_age'),
+    path('age/detail/<pk>/',AgeViewset.as_view({'get': 'retrieve'}), name='detail_age'),
+    path('age/update/<pk>/',AgeViewset.as_view({'put': 'update'}), name='update_age'),
+    path('age/delete/<pk>/',AgeViewset.as_view({'delete': 'destroy'}), name='delete_age'),
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/urls.py:40]: Need one more API get class between given age. i.e Given age 5 and find class for the age.

File: eCampusAPI/ecampus_api/master/views.py [hunk 3]
```diff
@@ -150,3 +150,26 @@ def perform_create(self, serializer):
...
+            return serializers.AgeCreateSerializer
+        if self.action == 'update':
+            return serializers.AgeUpdateSerializer
+        if self.action == 'retrieve':
+            return serializers.AgeDetailSerializer
+        if self.action == 'list':
+            return serializers.AgeDetailSerializer
+        return super(SectionViewset, self).get_serializer_class()
+
+    def perform_create(self, serializer):
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/views.py:170]: Condition Not required, only authenticated user will have access.

**Review by kishoreecampus — CHANGES_REQUESTED** [review]

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 4]
```diff
@@ -60,61 +60,61 @@ class Meta:
...
     class Meta:
-        model = Category
+        model = CasteCategory
         fields = ['category']
 
 class CategoryUpdateSerializer(serializers.ModelSerializer):
     class Meta:
-        model = Category
+        model = CasteCategory
         fields = ['category', 'created_on']
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:113]: Remove created_on

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 5]
```diff
@@ -60,61 +60,61 @@ class Meta:
...
-        fields = ['class_name', 'section_name', 'created_on']
+        fields = ['class_group','class_name', 'section_name', 'created_on']
     
 class SectionDetailSerializer(serializers.ModelSerializer):
     class Meta:
         model = Section
-        fields = ['id', 'class_name', 'section_name', 'created_on', 'created_by']  
+        fields = ['id','class_group', 'class_name', 'section_name', 'created_on', 'created_by']  
 
 class CategorySerializer(serializers.ModelSerializer):
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:100]: Check this

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 6]
```diff
@@ -138,3 +138,13 @@ class CasteDetailSerializer(serializers.ModelSerializer):
     class Meta:
         model = Caste
         fields = ['id', 'category', 'caste', 'created_on', 'created_by']
+
+class Anonymous_ClassNameDetailSerializer(serializers.ModelSerializer):
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:142]: Change Name

File: eCampusAPI/ecampus_api/master/views.py [hunk 7]
```diff
@@ -126,7 +126,7 @@ def perform_create(self, serializer):
         if self.request.user.is_authenticated:
             user = self.request.user.id
         else:
-            user = 0
+            user = 0
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/views.py:129]: Remove sapce

**Review by kishoreecampus — CHANGES_REQUESTED** [review]

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 5 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py:100]: yeah ajith Raghavendra sir asked us to get class group also in class section also

**Review by kishoreecampus — APPROVED** [review]

**Pull request merged**